### ISSUE TITLE
Supernote device does not connect and error when closing

### ISSUE THREAD

**Issue by amdemas** [nelinory/SupernoteDesktopClient#15]
I am on Windows 11 and the device does not show as connected when running the app. Also when closing I get the following error.

```
05/01/2023 15:03:21.721 [ERR] Fatal application exception: System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Text.Encodings.Web.OptimizedInboxTextEncoder.AllowedAsciiCodePoints.get_AsVector()
   at System.Text.Encodings.Web.OptimizedInboxTextEncoder.GetIndexOfFirstByteToEncodeSsse3(Byte* pData, UIntPtr lengthInBytes)
   at System.Text.Encodings.Web.OptimizedInboxTextEncoder.GetIndexOfFirstByteToEncode(ReadOnlySpan`1 data)
   at System.Text.Encodings.Web.DefaultJavaScriptEncoder.FindFirstCharacterToEncodeUtf8(ReadOnlySpan`1 utf8Text)
   at System.Text.Json.JsonWriterHelper.NeedsEscaping(ReadOnlySpan`1 value, JavaScriptEncoder encoder)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo.CacheNameAsUtf8BytesAndEscapedNameSection()
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo.Configure()
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo.InitializePropertyCache()
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo.Configure()
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo.<EnsureConfigured>g__ConfigureLocked|143_0()
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo.EnsureConfigured()
   at System.Text.Json.JsonSerializerOptions.GetTypeInfoInternal(Type type, Boolean ensureConfigured, Boolean resolveIfMutable)
   at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions options, Type inputType)
   at System.Text.Json.JsonSerializer.GetTypeInfo[T](JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at SupernoteDesktopClient.Core.SettingsManager.Save() in D:\My Projects\GitHub\SupernoteDesktopClient\Core\SettingsManager.cs:line 43
   at SupernoteDesktopClient.Views.Windows.MainWindow.OnClosing(Object sender, CancelEventArgs e) in D:\My Projects\GitHub\SupernoteDesktopClient\Views\Windows\MainWindow.xaml.cs:line 99
   at System.Windows.Window.OnClosing(CancelEventArgs e)
   at System.Windows.Window.WmClose()
   at System.Windows.Window.WindowFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by nelinory** [comment]
Hello @amdemas, I do not have a system running Windows 11, so it took me a couple of days until I was able to install Windows 11 under VMWare Workstation 17 Player.

Tested SDC with the following configuration:
- Windows 11 version 22H2 (OS Build 22621.1413)
- .NET 6 desktop: https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.16-windows-x64-installer?cid=getdotnetcore
- Latest SDC release: https://github.com/nelinory/SupernoteDesktopClient/releases/tag/v0.2.8514

Unfortunately I am not able to reproduce the error you are getting. On my Windows 11 test machine the SDC works as expected including the detection of Supernote device and ability to sync all notes. No errors were received under the multiple test scenarios executed.

A new version will be released next week, which have build in diagnostic mode with more detailed logging. Hopefully it helps us figuring out the issue you are having.
